src/crypto/sign.rs

```diff
@@ -0,0 +1,176 @@
+use crate::crypto::ed25519::{
+    ed25519_sign_internal, ed25519_verify_internal, ed25519_verifying_key_internal,
+};
+use bs58;
+use wasm_bindgen::prelude::*;
+
+/// Internal function to sign a message using Ed25519.
+/// - `message`: Raw bytes to sign
+/// - `secret`: Base58-encoded signing key with "signerSecret_z" prefix
+/// Returns base58-encoded signature with "signature_z" prefix or error string.
+pub(crate) fn sign_internal(message: &[u8], secret: &str) -> Result<String, String> {
+    let secret_bytes = bs58::decode(&secret["signerSecret_z".len()..])
+        .into_vec()
+        .map_err(|e| format!("Invalid base58 in secret: {:?}", e))?;
+
+    let signature = ed25519_sign_internal(&secret_bytes, message)
+        .map_err(|e| format!("Signing failed: {:?}", e))?;
+    Ok(format!(
+        "signature_z{}",
+        bs58::encode(signature).into_string()
+    ))
+}
+
+/// Internal function to verify an Ed25519 signature.
+/// - `signature`: Base58-encoded signature with "signature_z" prefix
+/// - `message`: Raw bytes that were signed
+/// - `id`: Base58-encoded verifying key with "signer_z" prefix
+/// Returns true if signature is valid, false otherwise, or error string if formats are invalid.
+pub(crate) fn verify_internal(signature: &str, message: &[u8], id: &str) -> Result<bool, String> {
+    if !signature.starts_with("signature_z") {
```

Comment:
mega nitpick: in some other places you use strip_prefix, which seems nicer

Reply:
Yeh I like that pattern too, all done